### PR TITLE
Add support for [Must][Un]MarshalOpt

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

**Comment by aboodman** [comment]
cc @arv

**Comment by aboodman** [comment]
TBR

File: go/marshal/encode.go [hunk 1]
```diff
@@ -166,6 +178,11 @@ func (e *marshalNomsError) Error() string {
 	return e.err.Error()
 }
 
+type Opt struct {
```
Comment (arv) [inline go/marshal/encode.go]: This could have used godoc too.

**Comment by phritz** [comment]
LGTM

**Pull request merged**